01_clean_services_from_csv.py

```diff
@@ -0,0 +1,99 @@
+"""
+01_clean_services_from_csv.py
+Cleans and deduplicates services from a CSV file.
+
+This script expects a CSV file with service/business data (e.g., from Yelp, Google Places, etc.)
+that may contain duplicate entries for the same business. It deduplicates by name and location,
+keeping the best rating when multiple entries exist for the same business.
+
+Expected input CSV columns:
+    - GEOID: Geographic identifier (string)
+    - name: Business/service name (string)
+    - rating: Rating value (float)
+    - lat: Latitude (float)
+    - lon: Longitude (float)
+    - category: Service category (string)
+
+Output: Cleaned CSV with one row per unique business (deduplicated by name and location)
+"""
+
+import argparse
+import os
+import pandas as pd
+import geopandas as gpd
+
+
+def clean_services(input_file, output_file):
+    """
+    Clean and deduplicate services from a CSV file.
+    
+    Args:
+        input_file: Path to input CSV file
+        output_file: Path to output CSV file
+    """
+    os.makedirs(os.path.dirname(output_file) if os.path.dirname(output_file) else ".", exist_ok=True)
+    
+    # read and clean the data
+    df = pd.read_csv(input_file, on_bad_lines="skip", dtype={
+        "GEOID": str, "name": str, "rating": float, "lat": float, "lon": float, "category": str
+    }).dropna(subset=["GEOID", "name", "lat", "lon"])
+    
+    for c in ["GEOID", "name", "category"]:
+        df[c] = df[c].astype(str).str.strip()
+    
+    # Deduplicate by name/location. Multiple entries may exist for the same business
+    # (e.g., from different data sources, different time periods, or data collection errors).
+    # When duplicates exist, we keep the maximum rating to ensure we retain the best
+    # representation of the business quality.
+    agg = (df.groupby(["name", "lat", "lon"], as_index=False)
+             .agg({
+                 "rating": "max",
+                 "GEOID":  lambda s: ",".join(sorted(set(map(str, s)))),
+                 "category": lambda s: ",".join(sorted(set([x for x in s if isinstance(x, str) and x]))),
```

Comment:
Why is it useful to concat categories in a single row?  E.g., say I get "grocery,pharmacy".  If later I'm trying to see from which places I can reach a grocery store within 30 minutes, won't I need to split it out again?